### PR TITLE
Added Linting, Scripts, Husky, Refactoring

### LINKED ISSUE #1632: Need for a linting setup, pre-commit hooks with husky for litmus portal
- Need linting with eslint, prettier, Airbnb Style Guide and typescript
- Scripts for linting and formatting for CI/CD pipeline
- Add husky and pre-commit hooks for code style enforcement 
- Format all existing components to the new style guide

### PR DESCRIPTION
**Pull request by arkajyotiMukherjee**
Signed-off-by: arkajyotiMukherjee <arkajyoti.mukherjee@mayadata.io>

**What this PR does / why we need it**:
Added linting with eslint, prettier, Airbnb SG and typescript; added scripts for linting and formatting; added husky and pre-commit hooks; formatted all components to the new SG

**Which issue this PR fixes** *(optional, in `fixes #<issue number>(, fixes #<issue_number>, ...)` format, will close that issue when PR gets merged)*: fixes #1632

**Checklist**

* [x] Does this PR have a corresponding GitHub issue?
* [ ] Have you included relevant README for the chaoslib/experiment with details?
* [ ] Have you added debug messages where necessary? 
* [x] Have you added task comments where necessary? 
* [ ] Have you tested the changes for possible failure conditions?
* [ ] Have you provided the positive & negative test logs for the litmusbook execution?
* [ ] Does the litmusbook ensure idempotency of cluster state?, i.e., is cluster restored to original state?
* [ ] Have you used non-shell/command modules for Kubernetes tasks?
* [ ] Have you (jinja) templatized custom scripts that is run by the litmusbook, if any? 
* [ ] Have you (jinja) templatized Kubernetes deployment manifests used by the litmusbook, if any?
* [ ] Have you reused/created util functions instead of repeating tasks in the litmusbook?
* [x] Do the artifacts follow the appropriate directory structure? 
* [ ] Have you isolated storage (eg: OpenEBS) specific implementations, checks? 
* [ ] Have you isolated platform (eg: baremetal kubeadm/openshift/aws/gcloud) specific implementations, checks?  
* [ ] Are the ansible facts well defined? Is the scope explicitly set for playbook & included utils?
* [ ] Have you ensured minimum/careful usage of shell utilities (awk, grep, sed, cut, xargs etc.,)?
* [ ] Can the limtusbook be executed both from within & outside a container (configurable paths, no hardcode)?
* [ ] Can you suggest the minimal resource requirements for the litmusbook execution?
* [ ] Does the litmusbook job artifact carry comments/default options/range for the ENV tunables?
* [ ] Has the litmusbooks been linted? 

**Special notes for your reviewer**:

### CODE REVIEW

File: litmus-portal/frontend/.eslintignore [hunk 1]
```diff
@@ -0,0 +1,2 @@
+**/node_modules/
+**/*.md
```
Comment (chandankumar4) [inline litmus-portal/frontend/.eslintignore:2]: Which .md file we ignoring here?

Comment (rajdas98) [reply litmus-portal/frontend/.eslintignore:2]: ignoring for the eslint

Comment (arkajyotiMukherjee) [reply litmus-portal/frontend/.eslintignore:2]: yes we are ignoring all md files for eslint

Comment (chandankumar4) [reply litmus-portal/frontend/.eslintignore:2]: ok

**Pull request merged**